core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -18,8 +18,32 @@
 #include "halt.h"
 #include "vmpu.h"
 #include "vmpu_freescale_k64_mem.h"
-
-#define MPU_MAX_REGIONS 11
+#include "page_allocator_faults.h"
+
+/* The K64F has 12 MPU regions, however, we use region 0 as the background
+ * region with `UVISOR_TACL_BACKGROUND` as permissions.
+ * Region 1 and 2 are used to unlock Application SRAM and Flash.
+ * Therefore 9 MPU regions are available for user ACLs.
+ * Region 3 and 4 are used to protect the current box stack and context.
+ * This leaves 6 MPU regions for round robin scheduling:
+ *
+ *     12      <-- End of MPU regions, K64F_MPU_REGIONS_MAX
+ * +---------+
+ * |   11    |
+ * |   ...   |
+ * |    5    | <-- Box Pages, K64F_MPU_REGIONS_USER
+ * +---------+
+ * |    4    | <-- Box Context
+ * |    3    | <-- Box Stack, K64F_MPU_REGIONS_RESERVED
+ * +---------+
+ * |    2    | <-- Application Flash unlock
+ * |    1    | <-- Application SRAM unlock
+ * |    0    | <-- Background region
+ * +---------+
+ */
+#define K64F_MPU_REGIONS_RESERVED 3
```

Comment:
@niklas-arm @AlessandroA please rename K64F_MPU_REGIONS_RESERVED to K64F_MPU_REGIONS_STATIC